site/docs/spec.md

```diff
@@ -929,6 +933,13 @@ This serialization scheme is for storing single values as individual binary valu
 | **`list`**                   | Not supported                                                                                                |
 | **`map`**                    | Not supported                                                                                                |
 
+## Appendix E: Default value Semantics
+
+Iceberg supports default-value semantics for fields of container
+types (i.e., struct, list and map). Specifically, a field of 
+type NestedField can have a default value that will be returned
```

Comment:
`NestedField` is a Java specific class, can we describe this generally within the spec instead?